### ISSUE TITLE
Warn when rename-refactoring to an existing identifier

### ISSUE THREAD

**Issue by gnl** [cursive-ide/cursive#1712]
Currently I can rename any identifier to an already existing one and inadvertently merge the semantic meanings, which makes it much harder to separate them again later on.

The only way to avoid this at the moment is to do a manual search before and that's an unnecessary hassle.

**Comment by cursive-ide** [comment]
Yes, this is tricky in Clojure. It's also ambiguous when vars or locals are intentionally shadowed - if I `declare` a var and then redefine it twice after that, and I rename it, should I just rename the actual definition, or all of them including the declare? It's not very clear what should be done.

You're right that this is almost certainly unexpected though, I'll add a warning for this.

**Comment by gnl** [comment]
Yes, I think it's fine not trying to be too smart about what the user might want, as that's not always clear, but a warning wouldn't break anything and help a lot with many use cases. Thanks!
